### ISSUE TITLE
another piece of valid html

### ISSUE THREAD

**Issue by Anniepoo** [SWI-Prolog/packages-http#4]
This validates, if wrapped in appropriate body & html tags, but fails to translate.

Markdown is chewing it up in the bug report, emailing a copy to Jan.

```
    {|html||
    <nav class="navbar navbar-default" role="navigation">
```

   <div class="navbar-header">
      <a class="navbar-brand" href="#">TutorialsPoint</a>
   </div>
   <div>
      <ul class="nav navbar-nav">
         <li class="active"><a href="#">iOS</a></li>
         <li><a href="#">SVN</a></li>
         <li class="dropdown">
            <a href="#" class="dropdown-toggle" data-toggle="dropdown">
               Java
               <b class="caret"></b>
            </a>
            <ul class="dropdown-menu">
               <li><a href="#">jmeter</a></li>
               <li><a href="#">EJB</a></li>
               <li><a href="#">Jasper Report</a></li>
               <li class="divider"></li>
               <li><a href="#">Separated link</a></li>
               <li class="divider"></li>
               <li><a href="#">One more separated link</a></li>
            </ul>
         </li>
      </ul>
   </div>
</nav>
        |}

**Comment by Anniepoo** [comment]
related to https://github.com/SWI-Prolog/packages-http/issues/1 ?

**Comment by wouterbeek** [comment]
Hi Annie, can you rephrase your question?  ATM I cannot distinguish the Github markdown from the HTML snippet you want to show us.  Also, what goes wrong exactly?  I assuming loading a text file containing a certain HTML snippet, e.g., with `load_html/3`?

**Comment by Anniepoo** [comment]
Apologies for the length of time this has been laying around.

The HTML quasiquoter should translate all valid HTML.
The above snippet can be surrounded by body and html tags and run through W3C's online validator.
But it will not be translated by the HTML quasiquoter.

If you get the snippet (stupid github markdown is tearing it up) from this file (or from me)
you'll see it doesn't get translated.

[quasiquoterproblem.txt](https://github.com/SWI-Prolog/packages-http/files/2611946/quasiquoterproblem.txt)

The issue is the **nav** tag. This tag is part of the new semantic markup in HTML5.
And we should actually conform to HTMLUnknownElement, which essentially says "If you don't know what a tag is, treat it as a div".

https://stackoverflow.com/questions/10830682/is-it-ok-to-use-unknown-html-tags

Given that one primary use for the HTML quasiquoter is to support adding bits of 
generated HTML, erring on the side of looseness rather than strictness seems wise.

**Comment by JanWielemaker** [comment]
Hmmm.  Can you be more complete?  The code loads and runs fine.  This was also my expectation as the HTML parser accepts unknown elements with a warning but the qq reader silences warnings AFAIK.

**Comment by Anniepoo** [comment]
Some time in the past apparently this has been fixed. So I'm closing.
